File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/NexusBuff.java
package tech.zolhungaj.amqapi.servercommands.objects;

import com.squareup.moshi.Json;

public record NexusBuff(

    @Json(name = "fileName")
    String filename,

    @Json(name = "debuff")
    boolean isDebuff,

    @Json(name = "name")
    String buffName,

    String description
) {}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/objects/NexusBuff.kt
package tech.zolhungaj.amqapi.servercommands.objects

import com.squareup.moshi.Json

@JvmRecord
data class NexusBuff(
    @Json(name = "localizationData") val localisationParameters : Map<String, String>?,
    @Json(name = "nameKey") val localisationNameKey: String,
    @Json(name = "descriptionKey") val localisationDescriptionKey: String,
    @Json(name = "fileName") val filename: String,
    @Json(name = "debuff") val isDebuff: Boolean,
    @Json(name = "name") val buffInternalName: String
)